### PR TITLE
Integrate lightning_utilities `is_overridden`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Follo-up to https://github.com/Lightning-AI/utilities/pull/35

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

File: src/pytorch_lightning/utilities/model_helpers.py [hunk 1]
```diff
@@ -11,12 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from functools import partial
-from typing import Any, Optional, Type
-from unittest.mock import Mock
-
-from lightning_utilities.core.imports import RequirementCache
-from torch import nn
+from typing import Optional, Type
```
Comment (carmocca) [inline src/pytorch_lightning/utilities/model_helpers.py:14]:
```suggestion
from typing import Any, Optional, Type

from lightning_utilities.core.imports import RequirementCache
from torch import nn
```

File: src/pytorch_lightning/utilities/model_helpers.py [hunk 2]
```diff
@@ -35,40 +29,6 @@ def is_overridden(method_name: str, instance: Optional[object] = None, parent: O
...
-        raise ModuleNotFoundError(str(_TORCHVISION_AVAILABLE))
-
-    from torchvision import models
-
-    torchvision_greater_equal_0_14 = RequirementCache("torchvision>=0.14.0")
-    # TODO: deprecate this function when 0.14 is the minimum supported torchvision
-    if torchvision_greater_equal_0_14:
-        return models.get_model(model_name, **kwargs)
-    return getattr(models, model_name)(**kwargs)
+    return is_overridden(method_name, instance, parent)
```
Comment (carmocca) [inline src/pytorch_lightning/utilities/model_helpers.py:34]:
```suggestion
    return is_overridden(method_name, instance, parent)


def get_torchvision_model(model_name: str, **kwargs: Any) -> nn.Module:
    from pytorch_lightning.utilities.imports import _TORCHVISION_AVAILABLE

    if not _TORCHVISION_AVAILABLE:
        raise ModuleNotFoundError(str(_TORCHVISION_AVAILABLE))

    from torchvision import models

    torchvision_greater_equal_0_14 = RequirementCache("torchvision>=0.14.0")
    # TODO: deprecate this function when 0.14 is the minimum supported torchvision
    if torchvision_greater_equal_0_14:
        return models.get_model(model_name, **kwargs)
    return getattr(models, model_name)(**kwargs)
```

**Review by tchaton — APPROVED** [review]
LGTM !
